Highway.cpp

```diff
@@ -22,8 +32,14 @@ void Highway::addVehicleInternal(Vehicle* v)
 
 void Highway::removeVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
+    //assert(false);
+
+    if( auto* car = dynamic_cast<Car*>(v) )
+        car->tryToEvade();
+    else if( auto* motorcycle = dynamic_cast<Motorcycle*>(v) )
+        motorcycle->lanesplitAndRace();
+    else if( auto* truck = dynamic_cast<SemiTruck*>(v) )
+        truck->tryToEvade();
```

Comment:
```
trucks pull over, but cars and bikes try to evade!!
```
- [ ] follow the instructions, add a `pullOver()` member function to `SemiTruck`